### DISCUSSION TITLE
upx installation error

### DISCUSSION THREAD

**Discussion by sikhilkrishnan** [Homebrew/discussions#1682] [Getting started]
Hi..
While installing upx in my MacAir M1 with mac OS Big Sur 11.4, I am getting below error
Error: upx: no bottle available!
Please help..

<img width="564" alt="Screenshot 2021-06-13 at 12 06 12 AM" src="https://user-images.githubusercontent.com/85804166/121786078-4886c400-cbdb-11eb-8a8d-135276496575.png">

**Comment by nandahkrishna** [comment]
`upx` isn't bottled on ARM because it isn't entirely supported yet (this may be fixed by a future release as the current one, `3.96`, is over a year old). There are some open issues at https://github.com/upx/upx/issues which you can check. Once the project supports ARM, it will be bottled.

Update: [This comment](https://github.com/upx/upx/issues/462#issuecomment-792127162) from the upstream project explains the current situation.

**Reply by sikhilkrishnan** [reply]
Thank u very much for the reply..
I need to use xforce keygen for installing Autocad. Can u pls suggest some other alternative ?

**Reply by SMillerDev** [reply]
Pay for software or use open source alternatives.

**Reply by sikhilkrishnan** [reply]
> Pay for software or use open source alternatives.

Thank you for the reply..

**Reply by alebcay** [reply]
You may find [FreeCAD](https://www.freecadweb.org/) or [LibreCAD](https://librecad.org) to be helpful.

**Comment by Euwizardbrazil** [comment]
sikhilkrishnan at this point,

You can install from other, just type:
brew install --build-from-source upx
or
arch -arm64 brew install --build-from-source upx

You mount the force dmg and copy the force.app to Desktop. After that you type on terminal:
sudo upx -d ~/Desktop/xf-adsk20.app/Contents/MacOS/x-force

It will ask your password.

So you can run the force from Desktop.

**Reply by natsiosti** [reply]
Does

**Reply by natsiosti** [reply]
Sikhilkrishnan, have you used [Euwizardbrazil]'s method ?

I am having the same problem with cracking AutoCAD with xfce keygen.

Euwizardbrazil says the following commands work as stated above,

"brew install --build-from-source upx
or
arch -arm64 brew install --build-from-source upx"

I wonder whether these work ,and need to be tried out
